esp/esdllib/esdl_script.cpp

```diff
@@ -1402,7 +1489,30 @@ class CEsdlTransformOperationTrace : public CEsdlTransformOperationWithoutChildr
         {
             if (m_test && !sourceContext->evaluateAsBoolean(m_test))
                 return false;
-            targetContext->trace(m_label, m_select, scriptContext->getTraceToStdout());
+            StringBuffer    content;
+            bool            isValue;
+            ITraceProducer& tracer = scriptContext->tracer();
+            if (targetContext->prepareTrace(m_select, content, isValue))
+            {
+                IDataMaskingProfileContext* masker = scriptContext->masker();
+                if (masker)
+                {
+                    if (isValue)
+                    {
+                        // TODO: need value type and optional mask style to mask a value
+                    }
+                    else
+                    {
+                        char* buffer = const_cast<char*>(content.str());
+                        if (!masker->maskContent("xml", buffer, 0, content.length()))
+                            tracer.log(MCuserInfo, "no embedded values were masked");
```

Comment:
It seems a little verbose to trace that we didn't have to mask anything in the following trace content.

Reply:
Agreed. It's a sanity check during development and debugging. Will remove before completion.